interface/admin.py

```diff
@@ -1,6 +1,49 @@
+from zipfile import ZipFile
+from tempfile import TemporaryDirectory
+from pathlib import Path
+
 from django.contrib import admin
+from django.db.models import F
+from django.http import FileResponse
+
 from interface.models import Course, Assignment, Submission
 
 admin.site.register(Course)
 admin.site.register(Submission)
-admin.site.register(Assignment)
+
+
+@admin.register(Assignment)
+class AssignmentAdmin(admin.ModelAdmin):
+    actions = ['download_submissions']
+
+    def download_submissions(self, request, _assignment):
+        if _assignment.count() != 1:
+            raise RuntimeError('Only one assignment can be selected')
+
+        assignment = _assignment[0]
+
+        subs = Submission.objects.filter(
+                assignment=assignment).order_by(F('timestamp').asc())
```

Comment:
I think you can simply write `.order_by('timestamp')`, no? And the `assignment` object should have a `submission_set` attribute which is already the filtered queryset you want, except for order_by.